docarray/document/mixins/multimodal.py

```diff
@@ -0,0 +1,134 @@
+                        for element in attribute:
+                            doc, attribute_type = cls._from_obj(element, sub_type)
+                            if attribute_type == AttributeType.DOCUMENT:
+                                attribute_type = AttributeType.ITERABLE_DOCUMENT
+                            elif attribute_type == AttributeType.NESTED:
+                                attribute_type = AttributeType.ITERABLE_NESTED
+                            else:
+                                raise ValueError(
+                                    f'Unsupported type annotation inside Iterable: {sub_type}'
+                                )
+                            chunk.chunks.append(doc)
+                        multi_modal_schema[key] = {
+                            'attribute_type': attribute_type,
+                            'type': f'{field.type._name}[{sub_type.__name__}]',
+                            'position': len(root.chunks),
+                        }
+                        root.chunks.append(chunk)
+                else:
+                    raise ValueError(f'Unsupported type annotation {field.type._name}')
+            else:
+                doc, attribute_type = cls._from_obj(attribute, field.type)
+                multi_modal_schema[key] = {
+                    'attribute_type': attribute_type,
+                    'type': field.type.__name__,
+                    'position': len(root.chunks),
+                }
+                root.chunks.append(doc)
+
+        # TODO: may have to modify this?
+        root.tags = tags
+        root._metadata['multi_modal_schema'] = multi_modal_schema
+
+        return root
+
+    def get_multi_modal_attribute(self, attribute: str) -> 'DocumentArray':
+        from docarray import DocumentArray
+
+        if 'multi_modal_schema' not in self._metadata:
+            raise ValueError(
+                'the Document does not correspond to a Multi Modal Document'
+            )
+
+        if attribute not in self._metadata['multi_modal_schema']:
+            raise ValueError(
+                f'the Document schema does not contain attribute {attribute}'
+            )
+
+        attribute_type = self._metadata['multi_modal_schema'][attribute][
+            'attribute_type'
+        ]
+        position = self._metadata['multi_modal_schema'][attribute].get('position')
+
+        if attribute_type in [AttributeType.DOCUMENT, AttributeType.NESTED]:
+            return DocumentArray([self.chunks[position]])
+        elif attribute_type in [
+            AttributeType.ITERABLE_DOCUMENT,
+            AttributeType.ITERABLE_NESTED,
+        ]:
+            return self.chunks[position].chunks
+        else:
+            raise ValueError(
+                f'Invalid attribute {attribute}: must a Document attribute or nested dataclass'
+            )
+
+    @classmethod
+    def _from_obj(cls, obj, obj_type) -> typing.Tuple['Document', AttributeType]:
```

Comment:
it is weird to need `obj` and `obj_type`, why not understand it here?

Reply:
we're inferring the object type from the type annotation, not from the object instance.
For example, in the case of ImageDocument, the actual type is a numpy array, while the type annotation is ImageDocument which is a typevar that also covers np.array

Reply:
Is `obj_type` parameter necessary? we can use `isinstance()` to validate the object type.